### PR TITLE
feat: Rate limiter

### PR DESCRIPTION
**Pull request by rephus**
Closes: 

### Description:

This only applies rate limiting to the new login_options endpoint

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/UserService.ts [hunk 1]
```diff
@@ -1180,9 +1180,16 @@ export class UserService {
             return {
                 showOptions: [openIdIssuer],
                 forceRedirect: true,
-                redirectUri: getRedirectUri(openIdIssuer),
+                redirectUri: new URL(
+                    `/api/v1${getRedirectUri(openIdIssuer)}`,
```
Comment (rephus) [inline packages/backend/src/services/UserService.ts:1184]: full url, not related with rate limit. I pushed here by mistake, we should merge it anyway

File: packages/backend/src/controllers/rateLimiter.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { RequestHandler } from 'express';
+import { rateLimit } from 'express-rate-limit';
+
+export const limiter = rateLimit({
+    windowMs: 60 * 1000, // 1 minute
+    limit: 100, // Limit each IP to 100 requests per `window` (per minute).
```
Comment (rephus) [inline packages/backend/src/controllers/rateLimiter.ts:6]: is this a good limit ? it seems too much for this single endpoint, but at the same time, we should be able to reuse this limit for other endpoints

**Comment by owlas** [comment]
@rephus we should handle this here: https://github.com/lightdash/lightdash-cloud/blob/f9cd7f3bd8def49332958c4b8d89aa31daf8cf87/platform-automation/src/main.ts#L94-L94

**Review by magnew — APPROVED** [review]
Nice. I know this is waiting on the other PR.

**Pull request closed without merging**